shared/coordinator/src/model.rs

```diff
@@ -181,17 +203,22 @@ pub struct LLM {
     pub architecture: LLMArchitecture,
     pub checkpoint: Checkpoint,
     pub data_type: LLMTrainingDataType,
-    pub data_location: LLMTrainingDataLocation,
+    pub data_locations: FixedVec<LLMTrainingDataLocation, { MAX_DATA_LOCATIONS }>,
```

Comment:
nit: this might be a breaking change in the coordinator's onchain layout, would be worth making sure the test introduced in this PR still passes:

https://github.com/PsycheFoundation/psyche/pull/124